core/modules/user/user.module

```diff
@@ -498,20 +498,20 @@ function user_is_blocked($name) {
 /**
  * Checks if a user has a role.
  *
- * @param int $rid
- *   A role ID.
+ * @param string $role_name
+ *   A string representing the role name.
  * @param User $account
  *   (optional) A user account. Defaults to the current user.
  *
  * @return bool
  *   TRUE or FALSE.
  */
```

Comment:
Could we add a `since` line:

```
@since 1.14.0 First parameter changed from $rid (legacy numeric identifier) to $role_name.
```

A `since` line indicating that previously it didn't work at all might also be helpful but I'm not sure we'd include documentation on it being broken directly inline.